engine/Shopware/Controllers/Backend/Order.php

```diff
@@ -894,6 +899,59 @@ public function createMailAction()
         ]);
     }
 
+    /**
+     * Retrieves all available mail templates
+     */
+    public function getMailTemplatesAction()
+    {
+        $limit = $this->Request()->getParam('limit', 100);
+        $offset = $this->Request()->getParam('start', 0);
+        $order = $this->Request()->getParam('sort', []);
+        $filter = $this->Request()->getParam('filter', []);
+
+        /** @var QueryBuilder $mailTemplatesQuery */
+        $mailTemplatesQuery = $this->getModelManager()->getRepository(Mail::class)->getMailsListQueryBuilder(
```

Comment:
Maybe try joining the `DocumentType`s using a `WITH` join.

Reply:
Not that easy actually because we needed to conatinate strings in the codition. That doesnt work in doctrine.